openverse_catalog/dags/provider_api_scripts/freesound.py

```diff
@@ -0,0 +1,296 @@
+        else:
+            should_continue = False
+    return item_count
+
+
+def _get_batch_json(
+        endpoint=ENDPOINT,
+        headers=None,
+        retries=RETRIES,
+        query_param=None
+):
+    if headers is None:
+        headers = HEADERS
+    response_json = delayed_requester.get_response_json(
+        endpoint,
+        retries,
+        query_param,
+        headers=headers
+    )
+    if response_json is None:
+        return None
+    else:
+        results = response_json.get("results")
+        return results
+
+
+def _process_item_batch(items_batch):
+    for item in items_batch:
+        item_meta_data = _extract_audio_data(item)
+        if item_meta_data is None:
+            continue
+        audio_store.add_item(**item_meta_data)
+    return audio_store.total_items
+
+
+def _extract_audio_data(media_data):
+    """Extracts meta data about the audio file
+    Freesound does not have audio thumbnails"""
+    try:
+        foreign_landing_url = media_data["url"]
+    except (TypeError, KeyError, KeyError):
+        return None
+    foreign_identifier = _get_foreign_identifier(media_data)
+    if foreign_identifier is None:
+        return None
+    audio_url, duration = _get_audio_info(media_data)
+    if audio_url is None:
+        return None
+    item_license = _get_license(media_data)
+    if item_license is None:
+        return None
+    creator, creator_url = _get_creator_data(media_data)
+    file_properties = _get_file_properties(media_data)
+    audio_set, set_url = _get_audio_set(media_data)
+
+    return {
+        'title': _get_title(media_data),
+        'creator': creator,
+        'creator_url': creator_url,
+        'foreign_identifier': foreign_identifier,
+        'foreign_landing_url': foreign_landing_url,
+        'audio_url': audio_url,
+        'duration': duration,
+        'license_info': item_license,
+        'meta_data': _get_metadata(media_data),
+        'raw_tags': media_data.get('tags'),
+        'audio_set': audio_set,
+        'set_url': set_url,
+        'alt_files': _get_alt_files(media_data),
+        'category': FREESOUND_CATEGORY,
+        **file_properties
+    }
+
+
+def _get_audio_set(media_data):
+    # set name, set thumbnail, position of audio in set, set url
+    set_name = None
+    set_url = media_data.get('pack')
+    if set_url is not None:
+        set_name = _get_set_name(set_url)
+    return set_name, set_url
+
+
+@functools.lru_cache(maxsize=1024)
+def _get_set_name(set_url):
+    response_json = delayed_requester.get_response_json(
+        set_url,
+        3,
+        query_params={'token': API_KEY},
+    )
+    name = response_json.get('name')
+    return name
+
+
+def _get_alt_files(media_data):
+    alt_files = [{
+        'url': media_data.get('download'),
+        'format': media_data.get('type'),
+        'filesize': media_data.get('filesize'),
+        'bit_rate': media_data.get('bitrate'),
+        'sample_rate': media_data.get('samplerate')
```

Comment:
Is this structure for `alt_files` documented anywhere? Since it's a JSON column, and this is basically it's type signature, is this enforced at all or could provider scripts set this incorrectly?